### ISSUE TITLE
Using @Component Functions in Spring Cloud Stream still broken in 3.2.8/v2021.0.5

### ISSUE THREAD

**Issue by fschlag** [spring-cloud/spring-cloud-function#956]
It's still not possible to use Functions/Consumers etc. defined as `@Component` in the latest release 3.2.8 (or spring-cloud-dependencies v2021.0.5).
There was a similar issue with `@Component` annotated Functions that was resolved in #925 by @olegz.

It's not possible to cast a Class to `ParametrizedType` as it is done in this line:

https://github.com/spring-cloud/spring-cloud-function/blob/b90dd7e1498c1f132709a651410ce018e79e3c81/spring-cloud-function-context/src/main/java/org/springframework/cloud/function/context/config/KotlinLambdaToFunctionAutoConfiguration.java#L160

The difference is, when using a Function Bean like 
```kotlin
@Bean
fun myTestConsumer(): (String) -> String {
    return {...}
}
```
the actual `Type` is `kotlin.jvm.functions.Function1<java.lang.String, java.lang.String>` which **can be** cast to `ParametrizedType`.

whereas a Component like
```kotlin
@Component
class MyTestConsumer : Function<String, String> {
    override fun apply(t: String): String {
        TODO("Not yet implemented")
    }
}
```
resolved to the actual class, in my example (see below) `class com.example.springcloudfunctiontest.MyTestConsumer` which **cannot** be cast to `ParametrizedType`

**Sample**
The sample I prepared for #925 is still valid and not working with the latest version. I am sorry that I did not tested it earlier.

https://github.com/fschlag/SpringCloudFunctionTest

**Comment by fschlag** [comment]
May relates to #940 as well

**Comment by fschlag** [comment]
Hi @olegz , I tested the latest changes using the Snapshot Release of `spring-cloud-dependencies:2021.0.6-20221115.024836-32` from repo.spring.io (which includes your latest change).

Unfortunately this did not fix the issue. The same error still occurs when casting  at this line
https://github.com/spring-cloud/spring-cloud-function/blob/ace6c16dd5f51cdd112181504c4fb205b96bdae5/spring-cloud-function-context/src/main/java/org/springframework/cloud/function/context/config/KotlinLambdaToFunctionAutoConfiguration.java#L160

The adjustments you've made regarding detecting the `functionType` did not work. At the line above the Type is still the Test Consumer `com.example.springcloudfunctiontest.MyTestConsumer`

**Comment by fschlag** [comment]
Hi @olegz, I am not sure if you saw my previous comment. But this is **not fixed** with your previous change same for #940.
